src/main/java/com/labsynch/labseer/service/LsThingServiceImpl.java

```diff
@@ -2389,50 +2386,56 @@ else if (query.getRecordedDateLessThan() != null){
+								firstItxPredicates.add(firstThingLabelEquals);
+							}else if (interaction.getOperator().equals("!=")){
+								Predicate firstThingLabelNotEquals = criteriaBuilder.notEqual(firstThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								firstItxPredicates.add(firstThingLabelNotEquals);
+							}else if(interaction.getOperator().equals("~")){
+								Predicate firstThingLabelLike = criteriaBuilder.like(criteriaBuilder.lower(firstThingLabel.<String>get("labelText")), '%' + interaction.getThingLabelText().toLowerCase() + '%');
+								firstItxPredicates.add(firstThingLabelLike);
+							}else if(interaction.getOperator().equalsIgnoreCase("like")){
+								Predicate firstThingLabelLike = criteriaBuilder.like(criteriaBuilder.lower(firstThingLabel.<String>get("labelText")), '%' + interaction.getThingLabelText().toLowerCase() + '%');
+								firstItxPredicates.add(firstThingLabelLike);
+							}else if(interaction.getOperator().equals("!~")){
+								Predicate firstThingLabelNotLike = criteriaBuilder.notLike(criteriaBuilder.lower(firstThingLabel.<String>get("labelText")), '%' + interaction.getThingLabelText().toLowerCase() + '%');
+								firstItxPredicates.add(firstThingLabelNotLike);
+							}else if(interaction.getOperator().equals(">")){
+								Predicate firstThingLabelGreaterThan = criteriaBuilder.greaterThan(firstThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								firstItxPredicates.add(firstThingLabelGreaterThan);
+							}else if(interaction.getOperator().equals(">=")){
+								Predicate firstThingLabelGreaterThan = criteriaBuilder.greaterThanOrEqualTo(firstThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								firstItxPredicates.add(firstThingLabelGreaterThan);
+							}else if(interaction.getOperator().equals("<")){
+								Predicate firstThingLabelLessThan = criteriaBuilder.lessThan(firstThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								firstItxPredicates.add(firstThingLabelLessThan);
+							}else if(interaction.getOperator().equals("<=")){
+								Predicate firstThingLabelLessThan = criteriaBuilder.lessThanOrEqualTo(firstThingLabel.<String>get("labelText"), interaction.getThingLabelText());
+								firstItxPredicates.add(firstThingLabelLessThan);
+							}else{
+								Predicate firstThingLabelLike = criteriaBuilder.like(criteriaBuilder.lower(firstThingLabel.<String>get("labelText")), '%' + interaction.getThingLabelText().toLowerCase() + '%');
+								firstItxPredicates.add(firstThingLabelLike);
+							}
 						}else{
-							Predicate firstThingLabelLike = criteriaBuilder.like(criteriaBuilder.lower(firstThingLabel.<String>get("labelText")), '%' + interaction.getThingLabelText().toLowerCase() + '%');
+							Predicate firstThingLabelLike = criteriaBuilder.like(firstThingLabel.<String>get("labelText"), '%' + interaction.getThingLabelText() + '%');
```

Comment:
Just noticed here that the analogous query above is case-insensitive but this one is case-sensitive. I think it'd make sense for this default "no operator specified" to be case-insensitive too.

Reply:
Yes, I will make this change.